File: web-client/src/enum/building/DepositTypeEnum.js
const DEPOSIT = 'DEPOSIT';
const RETURN = 'RETURN';
const USE = 'USE';

function getDisplayText(value) {
  if (!value) {
    return '';
  }

  if (value === 'DEPOSIT') {
    return '存款';
  }

  if (value === 'RETURN') {
    return '退还';
  }

  if (value === 'USE') {
    return '还款';
  }
}

export default {
  DEPOSIT,
  RETURN,
  USE,
  getDisplayText,
};

File: building-site-api/src/main/java/com/hozensoft/smartsite/enumeration/DepositTypeEnum.java
package com.hozensoft.smartsite.enumeration;

public enum DepositTypeEnum {

    DEPOSIT(1, "存款"),

    RETURN(2, "退还"),

    USE(3, "发放工资");

    private Integer value;

    private String description;

    DepositTypeEnum(int value, String description){
        this.value = value;
        this.description = description;
    }

    public int getValue() {
        return value;
    }

    public String getDescription() {
        return description;
    }

    public static DepositTypeEnum valueOf(int i) {
        for (DepositTypeEnum status : DepositTypeEnum.values()) {
            if (i == status.getValue()) {
                return status;
            }
        }

        throw new IllegalArgumentException("Unknow Status's value " + i);
    }
}
